examples/component-versioning/versioning-mechanism.md

```diff
@@ -0,0 +1,172 @@
+# Versioning mechanism
+
+## Prerequisite
+
+1. Make sure [`addon-oam-kubernetes-local`](https://github.com/crossplane/addon-oam-kubernetes-local) was installed.
+2. Before component versioning mechanism released in Crossplane, you can run `go run examples/containerized-workload/main.go`
+   instead of using crossplane. After this feature was released in crossplane, install crossplane will work.
+3. Make sure [`Simple Rollout`](https://github.com/oam-dev/catalog/tree/master/traits/simplerollouttrait) was installed for this demo.
+
+## Containing Trait with revisionEnabled and ApplicationConfiguration always using the latest component
+
+After Simple Rollout Trait installed, please make sure you have its trait definition:
+
+```shell script
+$ kubectl get traitdefinitions.core.oam.dev
+NAME                                         DEFINITION-NAME
+simplerollouttraits.extend.oam.dev           simplerollouttraits.extend.oam.dev
+```
+
+Simple Rollout is a trait with TraitDefinition as below:
+
+```yaml
+apiVersion: core.oam.dev/v1alpha2
+kind: TraitDefinition
+metadata:
+  name: simplerollouttraits.extend.oam.dev
+spec:
+  revisionEnabled: true
+  definitionRef:
+    name: simplerollouttraits.extend.oam.dev
+``` 
+
+You can notice that revisionEnabled flag is true.
+
+Step 1. Create OAM component
+
+```shell script
+$ kubectl apply -f examples/containerized-workload/sample_component.yaml
+component.core.oam.dev/example-component created
+``` 
+
+The first step is the same with [Component Mutable Demo](./component-mutable.md#ApplicationConfiguration-always-using-the-latest-component)
+
+Step 2. Create AppConfig
+
+```shell script
+$ kubectl apply -f examples/component-versioning/versioning-demo-app.yaml
+applicationconfiguration.core.oam.dev/example-appconfig created
+```
+
+ContainerizedWorkload was created with **the name of revisionName**.
```

Comment:
can you spell out the appConfig here too so that this document is self-contained

Reply:
done